### PR TITLE
Enable resource builder flattening in rollout status

### LINKED ISSUE #1223: kubectl rollout status deployment --selector="" does not work correctly
**What happened**:
`kubectl --timeout=10s rollout status deployment --selector="app=api"` timeouts after 10 seconds.

**What you expected to happen**:
I expect kubectl to return rollout status of 1 or more deployments matching the provided selector.

**How to reproduce it (as minimally and precisely as possible)**:
Create a deployment with any labels, in my case 'app=api'.
Try to get rollout status by using this command `kubectl --timeout=10s rollout status deployment --selector="app=api"`

**Anything else we need to know?**:
Verbose output shows that kubectl does a GET request to find out deployments with provided selector and gets a `DeploymentList` back, however second GET request does not use names of those deployments for followup query.
I can only conclude that this is a bug with kubectl.

**Environment**:
```
❯ k version
W0602 14:07:32.955543   71390 azure.go:92] WARNING: the azure auth plugin is deprecated in v1.22+, unavailable in v1.25+; use https://github.com/Azure/kubelogin instead.
To learn more, consult https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.1", GitCommit:"3ddd0f45aa91e2f30c70734b175631bec5b5825a", GitTreeState:"clean", BuildDate:"2022-05-24T12:17:11Z", GoVersion:"go1.18.2", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6", GitCommit:"07959215dd83b4ae6317b33c824f845abd578642", GitTreeState:"clean", BuildDate:"2022-03-30T18:28:25Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.24) and server (1.22) exceeds the supported minor version skew of +/-1
```

### PR DESCRIPTION
**Pull request by ardaguclu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
To make selector flag is working in `rollout status` command, we need to flatten resources in builder like it is done in other `rollout` commands (ie. (https://github.com/kubernetes/kubernetes/blob/c4f8c57b43843abd13bae677b73c622e1c30b46f/staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_history.go#L150)

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubectl/issues/1223

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Review by eddiezane — APPROVED** [review]
Thank you @ardaguclu! May I please ask you to add tests for rollout status with this PR?

File: staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go [hunk 1]
```diff
@@ -170,6 +165,7 @@ func (o *RolloutStatusOptions) Run() error {
 		ResourceTypeOrNameArgs(true, o.BuilderArgs...).
 		ContinueOnError().
 		Latest().
+		Flatten().
```
Comment (pacoxu) [inline staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go:168]: LGTM

**Review by pacoxu — APPROVED** [review]
Most scenarios are coved by test cases.

**Comment by ardaguclu** [comment]
@eddiezane @pacoxu would you mind tagging this PR?. Thanks.

**Pull request merged**